uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/b2c/GoogleLoginComponentHandler.java

```diff
@@ -0,0 +1,108 @@
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.interaction.b2c;
+
+import android.widget.EditText;
+
+import androidx.annotation.NonNull;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+
+import com.microsoft.identity.client.ui.automation.interaction.ILoginComponentHandler;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+
+import org.junit.Assert;
+
+/**
+ * A login component handler for Google IdP
+ */
+public class GoogleLoginComponentHandler implements ILoginComponentHandler {
+
+    @Override
+    public void handleEmailField(@NonNull final String username) {
+        UiAutomatorUtils.handleInput("identifierId", username);
+        handleNextButton();
+    }
+
+    @Override
+    public void handlePasswordField(@NonNull final String password) {
+        UiAutomatorUtils.handleInput("password", password);
+        final UiObject passwordBox = UiAutomatorUtils.obtainUiObjectWithResourceId("password");
+
+        try {
+            final UiObject passwordInput = passwordBox.getChild(
+                    new UiSelector().className(EditText.class)
+            );
+
+            passwordInput.setText(password);
+        } catch (UiObjectNotFoundException e) {
+            Assert.fail(e.getMessage());
+        }
+
+        handleNextButton();
+    }
+
+    @Override
+    public void handleBackButton() {
+        UiAutomatorUtils.pressBack();
+    }
+
+    @Override
+    public void handleNextButton() {
+        final UiObject nextBtn = UiAutomatorUtils.obtainUiObjectWithText("Next");
+        try {
+            nextBtn.click();
+        } catch (UiObjectNotFoundException e) {
+            Assert.fail(e.getMessage());
+        }
+    }
+
+    @Override
+    public void handleAccountPicker(@NonNull final String username) {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+
+    private UiObject getConsentScreen() {
+        return UiAutomatorUtils.obtainUiObjectWithResourceId("consentHeader");
+    }
+
+    @Override
+    public void confirmConsentPageReceived() {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+
+    @Override
+    public void acceptConsent() {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+
+    public void declineConsent() {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+
+    @Override
+    public void handleSpeedBump() {
+        throw new UnsupportedOperationException("Not supported for B2C Local Provider");
+    }
+}
```

Comment:
These messages need to say Google instead of Local

Reply:
Could they be in an abstract class that calls into an abstract getName() method so we don't have to write it in every case?

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/939/commits/8d08d2646d79e6e2d044a047b4ec44b2ba9f8f6c